### PR TITLE
Improvements - fixing problems, adding testing

### PR DESCRIPTION
**Pull request by nick-o**
Hi Guys,

I am looking at using the xDatabase resource but have found some problems with it:
- Test-TargetResource fails with InvokeMethodOnNull exception when not passing in Credentials (therefore using Integrated Authentication)
- Set-TargetResource fails with same above exception when not passing in Credentials
- Get-TargetResource was not implemented at all
- No Pester tests whatsoever exist for this resource

I have worked on the above items and this PR is addressing all of them. I guessed that the dev branch was the best branch to issue the PR against judging from the branching model image in CONTRIBUTING.md.

Let me know if you need anything else or if this is in fact not the correct branch.

Thanks
Nico

### CODE REVIEW

**Comment by vors** [comment]
Thank you Nick for your contribution. You are absolutely right about `dev` [branch purpose](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#lifecycle-of-a-pull-reqeust).

The fact that you are adding pester tests is awesome, my personal thanks for this.

In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.

File: Tests/MSFT_xDatabase.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    }
+    Context "database does exist" {
+
+        $connectionObj | Add-Member -MemberType ScriptMethod -Name Open -Value {return $null} -Force
+        It "Test-TargetResource should return false" {
+            Test-TargetResource @testParameter | Should Be $true
+        }
+    }
+}
```
Comment (vors) [inline Tests/MSFT_xDatabase.Tests.ps1]: New line at the end of file. http://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

**Comment by vors** [comment]
@Abhik11 it looks like we are already doing it, right? @nick-o just moved things around a little bit.

**Comment by Abhik11** [comment]
Yes, I did got confused by the diff. It looks OK.

File: Tests/MSFT_xDatabase.Tests.ps1 (hunk 1 above)
Comment (nick-o) [inline Tests/MSFT_xDatabase.Tests.ps1]: Oversight on my part...just added that, test pending

**Pull request merged**